### PR TITLE
add mu4e-maildirs-extension

### PR DESCRIPTION
**Pull request by choppsv1**
Add optional mu4e-maildirs extension to mu4e layer.

### CODE REVIEW

File: layers/+email/mu4e/README.org [hunk 1]
```diff
@@ -33,6 +34,20 @@ Configuration varies too much to give precise instructions.  What follows is one
 example configuration.  Refer to mu4e's manual for more detailed configuration
 instructions.
 
+** Maildirs extension
+The maildirs extension adds a list of all your maildirs to the main mu4e view
+that by default shows the unread and total mail counts for all your mail under
+your base mail directory.
+
+If you wish enable the maildirs extension you can set mu4e-use-maildirs to non
```
Comment (StreakyCobra) [inline layers/+email/mu4e/README.org]: Should be `=mu4e-use-maildirs=`

File: layers/+email/mu4e/packages.el [hunk 2]
```diff
@@ -38,4 +39,10 @@
 
       (when mu4e-account-alist
         (add-hook 'mu4e-compose-pre-hook 'mu4e/set-account)
-        (add-hook 'message-sent-hook 'mu4e/mail-account-reset)))))
+        (add-hook 'message-sent-hook 'mu4e/mail-account-reset))
+      (if mu4e-use-maildirs
```
Comment (StreakyCobra) [inline layers/+email/mu4e/packages.el]: It's cosmetic, but a `when` would be more appropriate and consistent with the rest of the code.

File: layers/+email/mu4e/packages.el [hunk 3]
```diff
@@ -38,4 +39,10 @@
...
+        (mu4e-maildirs-extension)))))
+
+(defun mu4e/init-mu4e-maildirs-extension ()
+  (use-package mu4e-maildirs-extension
+    :commands (mu4e-maildirs-extension)))
```
Comment (StreakyCobra) [inline layers/+email/mu4e/packages.el]: You can probably make the package loading conditional: `:if mu4e-use-maildirs`

Comment (choppsv1) [inline layers/+email/mu4e/packages.el]: So.. I was counting on the implicit :defer due to having a :commands. Are there situations where mu4e/init-mu4e-maidirs-extension is evaluated prior to the user setting the mu4-use-maildirs variable but before the mu4e :config is run?

Comment (StreakyCobra) [inline layers/+email/mu4e/packages.el]: I was unsure, you are probably right.

Comment (TheBB) [inline layers/+email/mu4e/packages.el]: Layer variables should be set together with the layer declaration in the dotfile, so they ought to be set at this point. I agree with @StreakyCobra that you can use `:if` here, and move the call to `mu4e-maildors-extension` as well (use `with-eval-load` or similar).

Comment (choppsv1) [inline layers/+email/mu4e/packages.el]: Done. Thanks.

**Comment by choppsv1** [comment]
Squashed commits, ready for merge! :)

File: layers/+email/mu4e/README.org [hunk 4]
```diff
@@ -33,6 +34,20 @@ Configuration varies too much to give precise instructions.  What follows is one
...
+that by default shows the unread and total mail counts for all your mail under
+your base mail directory.
+
+If you wish enable the maildirs extension you can set =mu4e-use-maildirs= to non
+nil value.
+
+#+begin_src emacs-lisp
+  (setq-default dotspacemacs-configuration-layers
+                '((mu4e :variables
+                        mu4e-use-maildirs t))
```
Comment (StreakyCobra) [inline layers/+email/mu4e/README.org]: Missing a closing parenthesis here :-)

Comment (choppsv1) [inline layers/+email/mu4e/README.org]: So I was.. Fixed now.

**Comment by choppsv1** [comment]
somehow messed up my branch.. next time i'll start using a one-off

**Pull request closed without merging**